### PR TITLE
feat: fetch wallet tokens metadata on load wallet (old facade)

### PR DESCRIPTION
**Pull request by pedroferreira1**
### Acceptance criteria

- We must have methods to load token metadata from explorer service (working for testnet or mainnet)

### CODE REVIEW

**Review by andreabadesso — APPROVED** [review]

**Review by andreabadesso — CHANGES_REQUESTED** [review]
> The wallet object must load the wallet tokens metadata and emit an event with it

The code looks good, but where is this event being emitted? Is this PR still a draft?

**Review by andreabadesso — APPROVED** [review]

File: src/api/metadataApi.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import explorerServiceAxios from './explorerServiceAxios';
+
+const metadataApi = {
+  async getDag(id: string, network: string) {
```
Comment (msbrogli) [inline src/api/metadataApi.ts:11]: I'd rename it to `getDagMetadata(...)`.

File: src/api/explorerServiceAxios.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * Returns the correct base url constant for wallet service based on the network
+ *
+ * @param {string} network The network, can be either mainnet or testnet
+ */
+const getBaseUrl = (network: string): string => {
+  if (network === 'mainnet') {
+    return EXPLORER_SERVICE_BASE_URL;
+  } else {
+    return EXPLORER_SERVICE_TESTNET_BASE_URL;
```
Comment (msbrogli) [inline src/api/explorerServiceAxios.ts:48]: Shouldn't we allow the client to set a different URL? This seems relevant for testing private networks, for example.

**Review by msbrogli — CHANGES_REQUESTED** [review]

**Review by msbrogli — APPROVED** [review]

File: src/api/explorerServiceAxios.ts (hunk 2 above)
Comment (pedroferreira1) [reply src/api/explorerServiceAxios.ts:48]: I agree with you but we already have another issue more generic to handle this kind of configuration (https://github.com/HathorNetwork/hathor-wallet-lib/issues/161)

I think we should leave this to another PR. What do you think?

File: src/api/metadataApi.ts (hunk 1 above)
Comment (msbrogli) [reply src/api/metadataApi.ts]: Thanks!

File: src/api/explorerServiceAxios.ts (hunk 2 above)
Comment (msbrogli) [reply src/api/explorerServiceAxios.ts:48]: I agree. Luis should work on it this week, right?

Comment (pedroferreira1) [reply src/api/explorerServiceAxios.ts:48]: Yes.

**Pull request merged**